File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = require("fs")
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n");

const N = +input[0];
const arr = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);
const X = +input[2];

let answer = 0;

for (let i = 0; i < N; i++) {
  let start = arr[i];
  let end = arr[arr.length - 1];

  while (start < end) {
    if (start + end === X) {
      answer++;
      arr.pop();
      break;
    } else if (start + end > X) {
      end = arr[arr.length - 2];
      arr.pop();
    } else if (start + end < X) {
      break;
    }
  }
}

console.log(answer);

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input());
arr = list(map(int, input().strip().split(" ")));
x = int(input());

arr.sort();


answer = 0;
left = 0;
right = n-1;


while left < right:
    cur_sum = arr[left] + arr[right];
    if cur_sum == x:
        answer += 1;
        left +=1;
    elif cur_sum < x:
        left+=1;
    elif cur_sum > x:
        right-=1;

print(answer);